File: app/src/main/java/xyz/michaelobi/paperplayer/data/model/Artist.java
package xyz.michaelobi.paperplayer.data.model;

import android.database.Cursor;
import android.provider.MediaStore;

/**
 * PaperPlayer Michael Obi 21 01 2017 8:35 AM
 */

public class Artist {

    private long id;
    private String name;
    private int numOfAlbums;
    private int numOfSongs;
    private String artistKey;

    public Artist(long id, String name, int numOfAlbums, int numOfSongs, String artistKey) {

        this.id = id;
        this.name = name;
        this.numOfAlbums = numOfAlbums;
        this.numOfSongs = numOfSongs;
        this.artistKey = artistKey;
    }

    public static Artist from(Cursor musicCursor) {
        int idColumn = musicCursor.getColumnIndex
                (MediaStore.Audio.Artists._ID);
        int artistColumn = musicCursor.getColumnIndex
                (MediaStore.Audio.Artists.ARTIST);
        int numOfSongsColumn = musicCursor.getColumnIndex
                (MediaStore.Audio.Artists.NUMBER_OF_TRACKS);
        int numOfAlbumsColumn = musicCursor.getColumnIndex
                (MediaStore.Audio.Artists.NUMBER_OF_ALBUMS);
        int artistKeyColumn = musicCursor.getColumnIndex
                (MediaStore.Audio.Artists.ARTIST_KEY);

        Artist artist = new Artist(musicCursor.getLong(idColumn),
                musicCursor.getString(artistColumn),
                musicCursor.getInt(numOfAlbumsColumn),
                musicCursor.getInt(numOfSongsColumn),
                musicCursor.getString(artistKeyColumn));
        return artist;
    }

    public int getNumOfSongs() {
        return numOfSongs;
    }

    public void setNumOfSongs(int numOfSongs) {
        this.numOfSongs = numOfSongs;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public int getNumOfAlbums() {
        return numOfAlbums;
    }

    public void setNumOfAlbums(int numOfAlbums) {
        this.numOfAlbums = numOfAlbums;
    }

    public String getArtistKey() {
        return artistKey;
    }

    public void setArtistKey(String artistKey) {
        this.artistKey = artistKey;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/xyz/michaelobi/paperplayer/data/model/Artist.kt
/*
 * MIT License
 *
 * Copyright (c) 2017 MIchael Obi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package xyz.michaelobi.paperplayer.data.model

import android.database.Cursor
import android.provider.MediaStore

/**
 * PaperPlayer Michael Obi 21 01 2017 8:35 AM
 */

class Artist(var id: Long, var name: String?, var numOfAlbums: Int, var numOfSongs: Int, var artistKey: String?) {
    companion object {
        fun from(musicCursor: Cursor): Artist {
            val idColumn = musicCursor.getColumnIndex(MediaStore.Audio.Artists._ID)
            val artistColumn = musicCursor.getColumnIndex(MediaStore.Audio.Artists.ARTIST)
            val numOfSongsColumn = musicCursor.getColumnIndex(MediaStore.Audio.Artists.NUMBER_OF_TRACKS)
            val numOfAlbumsColumn = musicCursor.getColumnIndex(MediaStore.Audio.Artists.NUMBER_OF_ALBUMS)
            val artistKeyColumn = musicCursor.getColumnIndex(MediaStore.Audio.Artists.ARTIST_KEY)

            val artist = Artist(musicCursor.getLong(idColumn),
                    musicCursor.getString(artistColumn),
                    musicCursor.getInt(numOfAlbumsColumn),
                    musicCursor.getInt(numOfSongsColumn),
                    musicCursor.getString(artistKeyColumn))
            return artist
        }
    }
}
